### PR TITLE
Add softmax to cls model

### PR DESCRIPTION
**Pull request by grimoire**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdeploy/codebase/mmcls/models/classifiers/base.py [hunk 1]
```diff
@@ -32,4 +33,6 @@ def base_classifier__forward(
     output = self.extract_feat(batch_inputs)
     if self.head is not None:
         output = self.head(output)
+
+    output = F.softmax(output, dim=1)
```
Comment (lvhan028) [inline mmdeploy/codebase/mmcls/models/classifiers/base.py:37]: For 'MultiLabelLinearClsHead', it uses 'sigmoid'

**Review by hanrui1sensetime — COMMENTED** [review]
There is still different result between executing `deploy.py` and python sdk demo. Maybe because of different preprocessing, but need to check.
For example:
```bash
python tools/deploy.py configs/mmcls/classification_onnxruntime_dynamic.py ../mmclassification/configs/resnet/resnet18_8xb32_in1k.py https://download.openmmlab.com/mmclassification/v0/resnet/resnet18_8xb32_in1k_20210831-fbbb1da6.pth ../mmclassification/demo/demo.JPEG --work-dir ./resnet-ort --dump-info
```
got 0.38 score with class 58 (water snake), 
while executing
```bash
./image_classification cpu ../../resnet-ort ../../../mmclassification/demo/demo.JPEG
```
got 0.31 score with class 58.

File: mmdeploy/codebase/mmcls/deploy/classification_model.py [hunk 2]
```diff
@@ -84,11 +84,21 @@ def forward(self,
...
                                   inputs})[self.output_names[0]]
 
-        from mmcls.models.heads.cls_head import ClsHead
-        predict = ClsHead._get_predictions(
-            None, cls_score, data_samples=data_samples)
-
-        return predict
+        from mmcls.structures import ClsDataSample
+        pred_scores = cls_score
+        pred_labels = pred_scores.argmax(dim=1, keepdim=True).detach()
```
Comment (lvhan028) [inline mmdeploy/codebase/mmcls/deploy/classification_model.py:89]: I found that in https://github.com/open-mmlab/mmclassification/blob/5b266d9e7c49044d1fcf15d046df3c06af3ff505/mmcls/models/heads/multi_label_cls_head.py#L140, `MultiLabelClsHead` doesn't call `pred_scores.argmax`.

BTW, can we suggest mmclassification to change `predict`  , like mmdet does, so that we can reuse their `add_pred_to_datasample`
```
results_list = self.bbox_head.predict(
            x, batch_data_samples, rescale=rescale)
batch_data_samples = self.add_pred_to_datasample(
            batch_data_samples, results_list)
````

**Comment by lvhan028** [comment]
> There is still different result between executing `deploy.py` and python sdk demo. Maybe because of different preprocessing, but need to check. For example:
> 
> ```shell
> python tools/deploy.py configs/mmcls/classification_onnxruntime_dynamic.py ../mmclassification/configs/resnet/resnet18_8xb32_in1k.py https://download.openmmlab.com/mmclassification/v0/resnet/resnet18_8xb32_in1k_20210831-fbbb1da6.pth ../mmclassification/demo/demo.JPEG --work-dir ./resnet-ort --dump-info
> ```
> 
> got 0.38 score with class 58 (water snake), while executing
> 
> ```shell
> ./image_classification cpu ../../resnet-ort ../../../mmclassification/demo/demo.JPEG
> ```
> 
> got 0.31 score with class 58.
This issue has been recorded in feishu project: https://project.feishu.cn/openmmlab/story/detail/3404740
It will be investigated in future, no in this PR

**Review by lvhan028 — APPROVED** [review]

**Review by hanrui1sensetime — APPROVED** [review]
LGTM.
